api/v1/health/server/server.go

```diff
@@ -15,20 +15,65 @@ import (
 	"net"
 	"net/http"
 	"os"
-	"os/signal"
 	"strconv"
 	"sync"
-	"sync/atomic"
 	"time"
 
+	"github.com/go-openapi/loads"
 	"github.com/go-openapi/swag"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/net/netutil"
-	"golang.org/x/sys/unix"
 
 	"github.com/cilium/cilium/api/v1/health/server/restapi"
+	"github.com/cilium/cilium/api/v1/health/server/restapi/connectivity"
+
 	"github.com/cilium/cilium/pkg/api"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+// Cell implements the cilium health API REST API server when provided
+// the required request handlers.
+var Cell = cell.Module(
+	"cilium-health-api-server",
+	"cilium health API server",
+
+	cell.Provide(newForCell),
 )
 
+type serverParams struct {
+	cell.In
+
+	Lifecycle  hive.Lifecycle
+	Shutdowner hive.Shutdowner
+	Logger     logrus.FieldLogger
+
+	GetHealthzHandler                 restapi.GetHealthzHandler
+	ConnectivityGetStatusHandler      connectivity.GetStatusHandler
+	ConnectivityPutStatusProbeHandler connectivity.PutStatusProbeHandler
+}
+
+func newForCell(p serverParams) (*Server, error) {
+	swaggerSpec, err := loads.Analyzed(SwaggerJSON, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to swagger spec: %w", err)
```

Comment:
nit: load?
```suggestion
		return nil, fmt.Errorf("failed to load swagger spec: %w", err)
```

Reply:
Will fix in a follow-up PR

Reply:
bit of a conflict with one of the commits in https://github.com/cilium/cilium/pull/24661 which we should be able to resolve between (I also fixed this over there).